### PR TITLE
Network detail screen

### PR DESCRIPTION
**Pull request by radeknovis**
Contains rerecorded snapshots for Xcode 15/ iOS 17

### CODE REVIEW

File: Sources/Web3ModalUI/Miscellaneous/CircleProgressView.swift [hunk 1]
```diff
@@ -1,140 +1,222 @@
-import UIKit
 import SwiftUI
+import UIKit
 
-public struct CircleProgressView: UIViewRepresentable {
-    
+public struct DrawingProgressView: UIViewRepresentable {
+    var shape: DrawingProgressUIView.ShapePath
     var color: Color
```
Comment (flypaper0) [inline Sources/Web3ModalUI/Miscellaneous/CircleProgressView.swift]: any reasons to use var?

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            .store(in: &disposeBag)
+        
+        Web3Modal.instance.sessionResponsePublisher
+            .receive(on: DispatchQueue.main)
+            .sink { response in
+                switch response.result {
+                case let .response(value):
+                    let stringResponse = try? value.get(String.self)
+                    
+                    DispatchQueue.main.async {
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift]: you receiving on main queue as I see. Why DispatchQueue async needed here?

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                            return
+                        }
+                        
+                        self.triedAddingChain = true
+                        
+                        Task {
+                            try? await self.addEthChain(from: from, to: chain)
+                        }
+                    } else {
+                        DispatchQueue.main.async {
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift]: same question

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        case .onAppear:
+            Task { @MainActor in
+                // Switch chain
+                await switchChain(chain)
+            }
+        case .didTapRetry:
+            
+            self.triedAddingChain = false
+            self.switchFailed = false
+            Task { @MainActor in
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift:94]: I'm not sure that switchChain will be executed on main thread. You should mark switchChain method with @MainActor ig

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            .sink { (event: Session.Event, _: String, _: Blockchain?) in
+                if event.name == "chainChanged" {
+                    
+                    guard let chainReference = try? event.data.get(Int.self) else {
+                        return
+                    }
+                    
+                    DispatchQueue.main.async {
+                        self.store.selectedChain = ChainPresets.ethChains.first(where: { $0.chainReference == String(chainReference) })
+                        self.router.setRoute(Router.AccountSubpage.profile)
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift]: strong reference

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        Web3Modal.instance.sessionResponsePublisher
+            .receive(on: DispatchQueue.main)
+            .sink { response in
+                switch response.result {
+                case let .response(value):
+                    let stringResponse = try? value.get(String.self)
+                    
+                    DispatchQueue.main.async {
+                        self.store.selectedChain = chain
+                        self.router.setRoute(Router.AccountSubpage.profile)
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailViewModel.swift]: strong reference

File: Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailView.swift [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+struct NetworkDetailView: View {
+    @EnvironmentObject var store: Store
+    
+    @ObservedObject var viewModel: NetworkDetailViewModel
+    
+    
+    var body: some View {
+        VStack {
+            content()
+                .onAppear {
```
Comment (flypaper0) [inline Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailView.swift:13]: may be .task{ } will be enough here? It also has async context so no need to create extra Tasks

Comment (flypaper0) [reply Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailView.swift:13]: but .task is executing on first view load only

Comment (radeknovis) [reply Sources/Web3Modal/Screens/ChainSwitch/NetworkDetail/NetworkDetailView.swift:13]: Yeah but it is also iOS 15+

**Review by alexander-lsvk — APPROVED** [review]

**Pull request merged**